howso/residuals.amlg

```diff
@@ -388,6 +389,25 @@
 			)
 		)
 
+		;if there are unique features whose residuals should be computed, remove them from 'features' and update
+		;unique_features_for_populating_output to update the output at the end since unique features don't have residuals
+		(if (size !uniqueNominalsSet)
+			(seq
+				(assign (assoc
+					unique_features_for_populating_output (filter (lambda (contains_index !uniqueNominalsSet (current_value))) features)
+				))
+
+				(if (size unique_features_for_populating_output)
+					(assign (assoc
+						features (filter (lambda (not (contains_index !uniqueNominalsSet (current_value)))) features)
```

Comment:
would something like `(indices (remove (zip features) (indices uniqueNominalsSet) ))` be better here?

Reply:
no because (indices) is o(n), (zip) is o(n), and the 2nd (indices) is also o(n).  that's 3 o(n) operations. whereas i have a (filter) which is just one o(n) and a (contains_index) which is a lookup.